sdk/identity/azure-identity/src/main/java/com/azure/identity/OnBehalfOfCredentialBuilder.java

```diff
@@ -0,0 +1,44 @@
+package com.azure.identity;
+
+public class OnBehalfOfCredentialBuilder extends AadCredentialBuilderBase<OnBehalfOfCredentialBuilder> {
+    private String clientSecret;
+
+    /**
+     * Sets the client secret for the authentication.
+     * @param clientSecret the secret value of the AAD application.
+     * @return An updated instance of this builder.
+     */
+    public OnBehalfOfCredentialBuilder clientSecret(String clientSecret) {
+        this.clientSecret = clientSecret;
+        return this;
+    }
+
+    /**
+     * Configures the persistent shared token cache options and enables the persistent token cache which is disabled
+     * by default. If configured, the credential will store tokens in a cache persisted to the machine, protected to
+     * the current user, which can be shared by other credentials and processes.
+     *
+     * @param tokenCachePersistenceOptions the token cache configuration options
+     * @return An updated instance of this builder with the token cache options configured.
+     */
+    public OnBehalfOfCredentialBuilder tokenCachePersistenceOptions(TokenCachePersistenceOptions
+                                                                          tokenCachePersistenceOptions) {
+        this.identityClientOptions.setTokenCacheOptions(tokenCachePersistenceOptions);
+        return this;
+    }
+
+    /**
+     * Configure the User Assertion Scope to be used for OnBehalfOf Authentication request.
+     *
+     * @param userAssertionScope the user assertion to be used for On behalf Of authentication flow
+     * @return An updated instance of this builder with the user assertion scope configured.
+     */
+    public OnBehalfOfCredentialBuilder userAssertionScope(UserAssertionScope userAssertionScope) {
+        this.identityClientOptions.userAssertion(userAssertionScope.getUserAssertion());
+        return this;
+    }
```

Comment:
oh I like this! I’ll copy this in JS

Reply:
this is the advanced OBO credential, right? with a new MSAL client per user assertion?

Reply:
this API is outdated and removed now.